### ISSUE TITLE
Filter request does not include any relationship/resource parameters

### ISSUE THREAD

**Issue by wize-wiz** [laravel/nova-issues#1964]
I noticed that the filters request dispatched here https://github.com/laravel/nova-js/blob/ef64432f24c68d12602d1b015abc954da3943b37/src/mixins/Filterable.js#L44-L47

is not sending any parameters regarding relationship/resource information. Is this done with a purpose?

In many situations I'm currently working in, a resource view-index shown in a relationship (many-to-many) shouldn't apply default filters if default values were assigned to the filter.

Use case:
I have tons of Webinars with 200 to 1000 participants. Most of them are in the past and they don't get shown in the webinar index-view. Filters used for the webinar resource have default values set to, for example, show only upcoming webinars with an active state. 

Now when viewing webinars from a relationship point-of-view, almost all webinars will not show up if these are in the past and default filters are applied.

So possibly I would query the request instance with the `viaRelationship()` method the trait `InteractsWithRelatedResources` offers, which `NovaRequest` implements, if relationship/resource parameters were passed on to the said request instance:

```php
public function filters(Request $request) {
   if($request->viaRelationship()) {
      // no filters at all
      return [];
   }

   ... some filters
}
```

**Comment by ragingdave** [comment]
Seconded here, as I am in a similar situation where I should need to hide a filter that is already being applied by the relation it's shown on.
For example:
Locations have many products. Products has a filter to scope to a specific location in it's index resource. Then on a specific location detail page, the products show already scoped to the location, but the filter still shows which allows the user to break the view by selecting a different location in the filter.

**Comment by wize-wiz** [comment]
@ragingdave Changes for a working version can be seen [here](https://github.com/wize-wiz/nova-js/commit/b4a9be374f1b99ee1356a7748cf9f0e2779c7bba)

**Comment by wize-wiz** [comment]
Closed, fixed with 2.9.0, #2005 was a duplicate but who cares.
